app/lib/persistenceGitbase/gitlabService.ts

```diff
@@ -0,0 +1,708 @@
+
+      const exactNameMatch = usernameCheck.some((user: any) => {
+        const userName = (user.username || '').toLowerCase();
+        return userName === baseUsername.toLowerCase();
+      });
+
+      if (exactNameMatch) {
+        for (let i = 1; i <= 9; i++) {
+          const tempUsername = `${baseUsername}${i}`;
+
+          const nameExists = usernameCheck.some((user: any) => {
+            const userName = (user.username || '').toLowerCase();
+            return userName === tempUsername.toLowerCase();
+          });
+
+          if (!nameExists) {
+            username = tempUsername;
+            usernameAvailable = true;
+            break;
+          }
+        }
+
+        if (!usernameAvailable) {
+          const timestamp = Date.now();
+          username = `${baseUsername}_${timestamp}`;
+        }
+      }
+
+      const newUser = await this.gitlab.Users.create({
+        email,
+        username,
+        password: this._generateRandomPassword(),
+        name: username,
+        skipConfirmation: true,
+      });
+
+      return newUser as unknown as GitlabUser;
+    } catch (error) {
+      const errorMessage = error instanceof Error ? error.message : 'Unknown error';
+      throw new Error(`Failed to create or find user: ${errorMessage}`);
+    }
+  }
+
+  async getProject(user: GitlabUser, projectName: string): Promise<GitlabProject> {
+    try {
+      const response = await fetch(
+        `${this.gitlabUrl}/api/v4/users/${user.id}/projects?owned=true&search=${encodeURIComponent(
+          projectName,
+        )}&per_page=100`,
+        {
+          headers: {
+            'PRIVATE-TOKEN': this.gitlabToken,
+          },
+        },
+      );
+
+      if (!response.ok) {
+        throw new Error(`API Failed: ${response.status} ${response.statusText}`);
+      }
+
+      const projects = (await response.json()) as GitlabProject[];
+
+      const project = projects.find((p: any) => p.name.toLowerCase() === projectName.toLowerCase());
+
+      if (!project) {
+        throw new Error(`Not Found Project '${projectName}'`);
+      }
+
+      return {
+        id: project.id,
+        name: project.name,
+        path_with_namespace: project.path_with_namespace,
+        default_branch: project.default_branch,
+        created_at: project.created_at,
+        updated_at: project.updated_at,
+        description: project.description,
+        visibility: project.visibility,
+      } as GitlabProject;
+    } catch (error) {
+      const errorMessage = error instanceof Error ? error.message : 'Unknown error';
+      throw new Error(`Failed to get or create project: ${errorMessage}`);
+    }
+  }
+
+  async createProject(user: GitlabUser, projectName: string, description?: string): Promise<GitlabProject> {
+    try {
+      const response = await fetch(
+        `${this.gitlabUrl}/api/v4/users/${user.id}/projects?owned=true&search=${encodeURIComponent(
+          projectName,
+        )}&per_page=100`,
```

Comment:
is it okay that we don't lookup next page when user have 100+ project?

Reply:
speedrun mode, I’ll fix things later if anyone complains 😺